yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -238,7 +238,7 @@ object YarnSparkHadoopUtil {
     if (Utils.isWindows) {
       escapeForShell("-XX:OnOutOfMemoryError=taskkill /F /PID %%%%p")
     } else {
-      "-XX:OnOutOfMemoryError='kill %p'"
+      "-XX:OnOutOfMemoryError='echo OnOutOfMemoryError; kill %p'"
```

Comment:
That looks pretty convincing then. Seems like it parses the commands and launches each with sh